### DISCUSSION TITLE
`fork:true` and `fork:only` query parameters behave inconsistently when calling `/search/code`

### DISCUSSION THREAD

**Discussion by EndBug** [community/community#64791] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hey folks, I'm using the `/search/code` endpoint of the REST API, and it looks like there was an undocumented breaking change in the behavior around forks.

The [endpoint docs](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-code) are telling me to look at [this page (Searching code (legacy))](https://docs.github.com/en/search-github/searching-on-github/searching-code) to build the `q` parameter.  
In that page, in the [Considerations for code search](https://docs.github.com/en/search-github/searching-on-github/searching-code#considerations-for-code-search) paragraph, it says:

> Code in [forks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks) is only searchable if the fork has more stars than the parent repository. Forks with fewer stars than the parent repository are not indexed for code search. To include forks with more stars than their parent in the search results, you will need to add fork:true or fork:only to your query. For more information, see "[Searching in forks](https://docs.github.com/en/search-github/searching-on-github/searching-in-forks)."

This means that if I add `fork:true` to my query, some fork results will be included.
What actually happens now is that **sometimes** `fork:true` behaves like `fork:only`, displaying ONLY forks.

I don't know why it only happens on some searches

#### [EndBug/add-and-commit](https://github.com/EndBug/add-and-commit)

```sh
$ gh api "/search/code?q=EndBug/add-and-commit" --jq '.total_count'
1312 

$ gh api "/search/code?q=EndBug/add-and-commit%20fork:true" --jq '.total_count'
2552

$ gh api "/search/code?q=EndBug/add-and-commit%20fork:only" --jq '.total_count'
2522
```

#### [MathieuSoysal/Javadoc-publisher.yml](https://github.com/MathieuSoysal/Javadoc-publisher.yml)

```sh
$ gh api "/search/code?q=MathieuSoysal/Javadoc-publisher.yml" --jq '.total_count'
75

$ gh api "/search/code?q=MathieuSoysal/Javadoc-publisher.yml%20fork:true" --jq '.total_count'
13

$ gh api "/search/code?q=MathieuSoysal/Javadoc-publisher.yml%20fork:only" --jq '.total_count'
13
```

In both instances something is wrong: `fork:true` and `fork:only` produce the same result, and that leads to incompatible results: as a rule of thumb, the result from `fork:true` should be equal to the sum of those from `fork:only` and the "normal search"

---

Reference issue: https://github.com/EndBug/api/issues/18

**Comment by wesnishio** [comment]
I am not sure about the problem though, this discussion was very useful to me as I was failing to search for a keyword in a forked repo and I did not know why... Thanks anyway.
